cmd/image-builder/upload.go

```diff
@@ -84,6 +85,16 @@ func uploaderForCmdAWS(cmd *cobra.Command) (cloud.Uploader, error) {
 	if err != nil {
 		return nil, err
 	}
+	targetArch, err := cmd.Flags().GetString("arch")
+	if err != nil {
+		return nil, err
+	}
+	if targetArch == "" {
+		// XXX: move this into the awscloud library, i.e.
+		// empty targetArch should default to curernt instead
+		// of leaving it empty
```

Comment:
This is what `awscloud` already does, I think? See: https://github.com/osbuild/images/blob/main/pkg/cloud/awscloud/uploader.go#L140-L142